File: lizard-web/api/models/op.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional


class OpSummary(BaseModel):
    op: str
    last_updated: Optional[str] = None
    total_hides: int
    avg_yield: Optional[float] = None
    predominant_class: Optional[str] = None


class ClassDistribution(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    class_name: str = Field(serialization_alias="class")
    count: int
    pct: float


class OpDetail(BaseModel):
    op: str
    total_hides: int
    avg_area: Optional[float] = None
    avg_yield: Optional[float] = None
    class_distribution: list[ClassDistribution] = []


class OpListResponse(BaseModel):
    items: list[OpSummary]
    total: int
    page: int
    limit: int
    pages: int


class DashboardPeriod(BaseModel):
    total_hides: int
    avg_yield: Optional[float] = None


class RecentOp(BaseModel):
    op: str
    date: str
    total_hides: int


class DashboardResponse(BaseModel):
    today: DashboardPeriod
    week: DashboardPeriod
    today_class_distribution: list[ClassDistribution] = []
    recent_ops: list[RecentOp] = []

File: lizard-web/frontend/src/types/op.ts
export interface ClassDistribution {
  class: string
  count: number
  pct: number
}

export interface DashboardPeriod {
  total_hides: number
  avg_yield: number | null
}

export interface RecentOp {
  op: string
  date: string
  total_hides: number
}

export interface DashboardData {
  today: DashboardPeriod
  week: DashboardPeriod
  today_class_distribution: ClassDistribution[]
  recent_ops: RecentOp[]
}

export interface OpSummary {
  op: string
  last_updated: string | null
  total_hides: number
  avg_yield: number | null
  predominant_class: string | null
}

export interface OpDetail {
  op: string
  total_hides: number
  avg_area: number | null
  avg_yield: number | null
  class_distribution: ClassDistribution[]
}
